### PR TITLE
[rllib] Add examples page, add hierarchical training example, delete SC2 examples

### LINKED ISSUE #3713: [rllib] should add examples list to documentation
Currently the directory of examples is organized in an adhoc way. Add a documentation section that provides an overview of relevant examples for different parts of the API, similar to the tune examples page.

Also, add links to third party examples such as in flow and sagemaker.

### LINKED ISSUE #3856: [rllib] IMPALA creates not-needed environments
### Describe the problem
Every agent has local evaluator + wokers_num of remote evaluators. Each evaluator creates num_envs_per_worker environments, even the local one. In IMPALA training, local evaluator is used for optimizer and it does not use environments, so there is num_envs_per_worker too much. This can be a problem, as environments can use resources, as it is an example with SC2 environment we're using (each env takes ~0.5gb of RAM).

### Source code / logs
The mentioned code is in policy evaluator in the constructor in https://github.com/ray-project/ray/blob/master/python/ray/rllib/evaluation/policy_evaluator.py. All environments (local/remote) are treated equally. Local/remote evaluators are created in _make_evaluator at https://github.com/ray-project/ray/blob/master/python/ray/rllib/agents/agent.py.

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

(see title)

## Related issue number

Closes https://github.com/ray-project/ray/issues/3713
Adds documentation for to close https://github.com/ray-project/ray/issues/3856

### CODE REVIEW

**Comment by pcmoritz** [comment]
Does it make sense to put the starcraft environment into contrib (just to keep it around), or should it be removed entirely?

**Comment by ericl** [comment]
We should remove it once it's merged into the smac repo, basically it
doesn't make sense to maintain these sort of things in the Ray repo when
there is a project outside that is more well targeted etc.

File: python/ray/rllib/examples/hierarchical_training.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+from gym.spaces import Box, Discrete, Tuple
+import logging
+
+import ray
+from ray.tune import run_experiments, function
+from ray.rllib.env import MultiAgentEnv
+from ray.rllib.agents.ppo import PPOAgent
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--flat", action="store_true")
```
Comment (richardliaw) [inline python/ray/rllib/examples/hierarchical_training.py]: can you document a "help" for this?

**Review by richardliaw — APPROVED** [review]
Looks good.

One thing we need to do for Tune and RLlib is actually have example walkthroughs - like the A3C example. It will make things more accessible and understandable.

**Pull request merged**